src/zabapgit_syntax_highlighter.prog.abap

```diff
@@ -0,0 +1,762 @@
+              <prev>-offset = <prev>-offset + 1.                    " Shifted { out of highlight
+              CLEAR lv_prev_token.
+            ELSEIF <match>-text_tag = '{'.
+              <prev>-length = <match>-offset - <prev>-offset.
+              CLEAR lv_prev_token.
+            ELSEIF <prev>-text_tag = '}'.
+              <prev>-length = <match>-offset - <prev>-offset.
+              <prev>-offset = <prev>-offset + 1.                    " Shifted } out of highlight
+              CLEAR lv_prev_token.
+            ENDIF.
+            DELETE ct_matches INDEX lv_index.
+            CONTINUE.
+          ENDIF.
+      ENDCASE.
+
+      lv_prev_token = <match>-token.
+      ASSIGN <match> TO <prev>.
+    ENDLOOP.
+
+    " Add entries refering to parts of text that should not be formatted
+    LOOP AT ct_matches ASSIGNING <match>.
+      IF <match>-offset > lv_last_pos.
+        lv_length = <match>-offset - lv_last_pos.
+        ls_match-token  = c_token-none.
+        ls_match-offset = lv_last_pos.
+        ls_match-length = lv_length.
+        INSERT ls_match INTO ct_matches INDEX sy-tabix.
+      ENDIF.
+      lv_last_pos = <match>-offset + <match>-length.
+    ENDLOOP.
+
+    " Add remainder of the string
+    IF lv_line_len > lv_last_pos.
+      lv_length = lv_line_len - lv_last_pos.
+      ls_match-token  = c_token-none.
+      ls_match-offset = lv_last_pos.
+      ls_match-length = lv_length.
+      APPEND ls_match TO ct_matches.
+    ENDIF.
+
+  ENDMETHOD.  " order_matches.
+
+  METHOD format_line.
+
+    DATA:
+          lv_chunk     TYPE string,
+          lv_css_class TYPE string.
+
+    FIELD-SYMBOLS:
+          <match> TYPE ty_match.
+
+    LOOP AT it_matches ASSIGNING <match>.
+      lv_chunk = substring( val = iv_line off = <match>-offset len = <match>-length ).
+
+      CASE <match>-token.
+        WHEN c_token-keyword.
+          lv_css_class = c_css-keyword.
+        WHEN c_token-comment.
+          lv_css_class = c_css-comment.
+        WHEN c_token-text.
+          lv_css_class = c_css-text.
+        WHEN c_token-none.
+          CLEAR: lv_css_class.
+      ENDCASE.
+
+      lv_chunk = me->apply_style( iv_line  = lv_chunk
+                                  iv_class = lv_css_class ).
+
+      rv_line = rv_line && lv_chunk.
+    ENDLOOP.
+
+  ENDMETHOD.                    "format_line
+
+  METHOD apply_style.
+
+    DATA lv_escaped TYPE string.
+
+    lv_escaped = escape( val = iv_line  format = cl_abap_format=>e_html_attr ).
+    IF iv_class IS NOT INITIAL.
+      rv_line = |<span class="{ iv_class }">{ lv_escaped }</span>|.
+    ELSE.
+      rv_line = lv_escaped.
+    ENDIF.
+
+  ENDMETHOD.                    "apply_style
+
+  METHOD process_line.
+
+    DATA: lt_matches TYPE ty_match_tt.
+
+    IF strlen( iv_line ) = 0.
+      RETURN.
+    ENDIF.
+
+    lt_matches = me->parse_line( iv_line ).
+
+    me->order_matches( EXPORTING iv_line    = iv_line
```

Comment:
Hi @EduardoCopat ,
Shortly the idea behind the solution is as follows: 
- parse_line() - returns the table with all matches for texts, key words and comments.
- order_matches() - processes this table - removes some matches, which are not necessary (e.g. key words inside code that is commented out), fine-tunes the values of some matches and adds additional lines which represent plain code, which should not be formatted. At the end, each line in ct_matches represent exact part of the string and details necessary for formatting (starting position + length + reference to style);
- format_line() is quite simple - processes the table entry by entry and passes it to apply_style(). 

Hope that answers your question.